### ISSUE TITLE
Accessibility Testing for [vfs-mhv-secure-messaging, MHV- Secure Messaging, MHV-SecureMessaging]

### ISSUE THREAD

**Issue by bryan-riley-va** [department-of-veterans-affairs/va.gov-team#57585]
### Product information

- [X] Team name, product name, and feature name have been added to the title of this issue.
- [X] Team label, product label and feature label (if applicable) have been added to this issue.

### Use of color and color contrast checks

- [x] All text of 20px or smaller has a 4.5:1 contrast ratio to its background (or better)
- [x] All text of 20px or larger has a 3:1 contrast ratio to its background (or better)
- [x] Non-text elements have a 3:1  contrast ratio to their background and to neighboring elements (or better)
- [x] Color is not the only way to distinguish links from other text (eg. links are underlined)
- [x] Any charts, maps, infographics, and tables convey all information without only relying on color
- [x] Content does not refer to color, especially when providing user instructions (eg. "Click the blue button")

### How did color testing go?
Color testing found no issues

### Who completed the axe scans?

Raji Venkatesh 

### axe checks

- [x] Each page has been scanned using axe (results shared below)
- [x] axe is integrated into your end-to-end testing

### axe DevTools scan results

15 total Issues, 13 Issues are best practice violations.  Two are serious:  
Search Results Inbox  ul and ol must only directly contain li,script, or template elements
Search Results Drafts  ul and ol must only directly contain li,script, or template elements
(Please Request Word Doc Axe DevTools Scan Results.docx Attached From Raji Venkatesh for more details )
-See comment under ticket - these two serious issues are part of library component (global)

### Who completed the content zoom and reflow test?

Robert Stratton

### Content zoom and reflow checks

- [x] All page elements are readable and usable at 200% zoom
- [x] All page elements are readable and usable at 300% zoom
- [x] All page elements are readable and usable at 400% zoom

### How did content zoom and reflow testing go?

On the original testing, the Details Page Buttons get cut off vertically 200% zoom, but not at 300% and 400%, otherwise zoom is fine.  We completed MHV-44668 to resolve the zoom issue by:
-Increase a breakpoint, mobile view starts with a wider screen width
-On devices with a really small size screen, stack action buttons up in a columns instead of a row

### Who completed the keyboard navigation test?

Aaron Read

### Keyboard navigation checks

- [x] Each link, button, form input, checkbox, radio button, select menu, and custom element can receive keyboard focus
- [x] Each link, button, form input, checkbox, radio button, select menu, and custom element responds to expected keys
- [x] All elements under focus have a visible focus indicator
- [x] The order of [Tab] stops made sense and was appropriate for completing tasks

### How did keyboard testing go?

All keyboard navigation is compliant

### Do you have any other results to share?
-During our screen reader testing, we feel we have implemented the requested outline and header improvements (some on page and some hidden) as requested.  While we are hopeful our implementation of this meets the need and would not be a blocker, we have identified what we feel is a better implementation that we would like to pursue.   We feel we can use landmarks to more clearly guide a screen reader user through the pages more efficiently.  For example the inbox and reply draft pages are two examples we feel we could improve on (while also improving throughout the application).

We have been providing updates that we have made to the application in two different tickets:

First - the Midpoint review ticket listed MUST and SHOULD feedback and we have responded to this:
Midpoint Review ticket: https://github.com/department-of-veterans-affairs/va.gov-team/issues/51115
Accessibility Feedback from Midpoint: https://github.com/department-of-veterans-affairs/va.gov-team/issues/51913 - **51913 all MUST issues have been resolved - please review**

Second - the Accessibility team swarmed around additional findings and our comments and resolutions to these findings are in this ticket:  https://github.com/department-of-veterans-affairs/va.gov-team/issues/54823 - **Issues 8/9 are in progress**

**Comment by sara-amanda** [comment]
## CAIA Update
Hi, @bryan-riley-va replying to your Slack message here with links that you may find useful as you work towards completing this ticket to prep for staging review. Be sure to include any relevant images that need referenced in your findings. cc: @briandeconinck, @AngelaFowler82, @SarahKay8 

### Foundational Accessibility Testing (required)

The items mentioned in the ticket above, are linked with instructions in the checklist below. It looks like you completed the color and contrast portion, so I went ahead and marked it as done. 

Foundational accessibility testing will detect common issues and can be completed without a deep background in accessibility or the use of assistive technology.

- [x] [Use of color and color contrast](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review#use-of-color)
- [ ] [Automated testing with axe by Deque](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review#axe)
- [ ] [Content zoom and reflow](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review#content-zoom)
- [ ] [Keyboard navigation](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review#keyboard-nav)

### Advanced Tests (recommended)
Advanced accessibility testing detects accessibility barriers that may be missed by foundational testing, and may require more background knowledge or the use of assistive technology. (@AngelaFowler82 @SarahKay8 or myself @sara-amanda  can help if you get stuck here). 
- [ ] [WAVE spot checks](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review#wave)
- [ ] [Code quality review](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review#code-quality)
- [ ] [Mouse-only and touchscreen](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review#mouse-and-touch)
- [ ] [Screen readers](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review#screen-readers)
- [ ] [Voice commands](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review#voice)

## Next Steps
- If you have any other questions about the testing artifact, we can answer those, as you move through the artifact.
- **Let us know when your testing artifact is completed.** 
- **We can review and verify your tests, once the artifact is completed.** 
- You can always post questions in the [#accessibility-help](https://dsva.slack.com/archives/C8E985R32) channel too.
- Feel free to also attend [office hours](https://depo-platform-documentation.scrollhelp.site/support/office-hours) for the Collaboration Cycle.

**Comment by oleksii-morgun-va** [comment]
> Search Results Inbox ul and ol must only directly contain li,script, or template elements
> Search Results Drafts ul and ol must only directly contain li,script, or template elements
The following issues are related to `<va-pagination> `component 
<img width="200" alt="image" src="https://user-images.githubusercontent.com/87077843/235236908-5d9ba68c-ede9-4bb0-8629-24f925bebab7.png">

**Comment by oleksii-morgun-va** [comment]
Zoom issues were addressed here 
https://github.com/department-of-veterans-affairs/vets-website/pull/24116

**Comment by sara-amanda** [comment]
Closing as new tickets will be opened to address staging review.

**Comment by RLHecht** [comment]
Thanks, Sara. Closing!
